### PR TITLE
Generalised the reading for any Abaqus version.

### PR DESCRIPTION
**Pull request by tmnp19**
- Removed the `job_name` from the `.yaml` file since a Abaqus input file only has one job;
- Made the reading and comparison with Abaqus insensitive to upper and lower cases;
- Added a flag to support the reading of strings with spaces in the `.yaml` file and successfully find them in Abaqus `.inp` file.

### CODE REVIEW

File: piglot/solver/abaqus/reader.py [hunk 1]
```diff
@@ -200,15 +200,100 @@ def write_output_file(i, variables_array, variable, step, location, file_name):
...
+    ValueError
+        Raises an error if the set name is not found in the output database.
+    """
+    set_names_list = list(odb.rootAssembly.instances[instance].nodeSets.keys())
+    set_names_list_upper = [setName.upper() for setName in set_names_list]
+    set_name_upper = variables["set_name"].upper()
+    if set_name_upper not in set_names_list_upper:
+        raise ValueError("Set name not found in the output database.")
+    return set_names_list[set_names_list_upper.index(set_name_upper)]
+
```
Comment (ruicoelhopedro) [inline piglot/solver/abaqus/reader.py:291]: This block seems repeating. Consider generalising the case-insensitive searching for a key in a list of strings to a separate function, and then using it to get the step, instance and nodeset names.

File: piglot/solver/abaqus/reader.py [hunk 2]
```diff
@@ -219,12 +304,14 @@ def main():
     odb = openOdb(path=odb_name)
 
     # Create a variable that refers to the respective step
-    step = odb.steps[variables["step_name"]]
+    step = get_step(odb, variables)
+    instance_name = get_instance(odb, variables)
+    nodeset_name = get_set(odb, variables, instance_name)
```
Comment (ruicoelhopedro) [inline piglot/solver/abaqus/reader.py:309]: The new function should be used here afterwards, to fetch from the dicts

**Review by ruicoelhopedro — CHANGES_REQUESTED** [review]
Thank you for the work, this will make the interface more robust.

I've left some comments regarding a couple of refactorings to simplify the code.

File: piglot/solver/abaqus/reader.py (hunk 1 above)
Comment (tmnp19) [reply piglot/solver/abaqus/reader.py:291]: Done. Thank you for the suggestion.

**Review by ruicoelhopedro — APPROVED** [review]
Thanks, it's looking good. I'll merge this

**Pull request merged**